src/StreamdeckTypes/Sent/BaseStateType.ts

```diff
@@ -8,6 +8,7 @@ export const BaseStateProperties = {
   device: Type.String(),
   event: Type.RegEx(/^willAppear|willDisappear|didReceiveSettings$/),
   payload: Type.Object({
+    controller: Type.Optional(Type.RegEx(/^Keypad|Encoder$/)),
```

Comment:
This is really only present on the `willAppear` event and the new dial events, but it seems like it could be expanded in the future. If there's a better place for this, please let me know.

Reply:
Yeah, this is going to be a bit of an interesting one as the diagram of all events is getting a bit... complex. Let me do some more research on this one and see if I can find a way to clean this up.

![image](https://user-images.githubusercontent.com/5192145/206927891-e3d5dd23-3015-4d39-bf4b-9e37e946d3d9.png)

Reply:
I found the controller only in these two events as well.
Looks good 👍